apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -417,8 +468,41 @@ const EventTypePage = (props: EventTypeSetupProps) => {
       customInputs,
       children,
     });
+
+    setUnsavedDialogOpen(false);
   };
 
+  // Warning users for unsaved changes
+  useEffect(() => {
+    const handleBeforeChange = (url: string) => {
+      const currentUrl = router.asPath;
+      const newUrl = url;
+
+      const urlChanged = currentUrl.split("?")[0] === newUrl.split("?")[0];
+      // Check if the query parameters have changed
+
+      if (shouldWarnUnsaved && !urlChanged) {
+        const unsavedChanges = getObjectDifference({ ...formMethods.getValues() }, defaultValues);
+
+        if (unsavedChanges["availability"]) {
+          delete unsavedChanges["availability"];
+        }
+
+        if (Object.keys(unsavedChanges).length > 0) {
+          setUnsavedDialogOpen(true);
+          setNewRoute(url);
+          router.events.emit("routeChangeError");
+          throw "navigation aborted";
+        }
+      }
+    };
+    router.events.on("routeChangeStart", handleBeforeChange);
```

Comment:
i am not sure if we should use router.events because the new next router doesn't support route events